airbyte-integrations/connectors/source-mixpanel/source_mixpanel/manifest.yaml

```diff
@@ -0,0 +1,376 @@
+      datetime: '{{ config.end_date or day_delta(-1, format="%Y-%m-%dT%H:%M:%SZ") }}'
+      datetime_format: "%Y-%m-%dT%H:%M:%SZ"
+
+  # https://developer.mixpanel.com/reference/cohorts
+  cohorts_stream:
+    $ref: "#/definitions/stream_base"
+    $parameters:
+      name: cohorts
+      path: 2.0/cohorts/list
+      field_path: []
+    retriever:
+      $ref: "#/definitions/retriever"
+      record_selector:
+        $ref: "#/definitions/selector_empty_dpath"
+        record_filter:
+          condition: "{{ record['created'] >= stream_state.created if stream_state.created else true }}"
+    incremental_sync:
+      type: DatetimeBasedCursor
+      cursor_field: created
+      cursor_datetime_formats:
+        - "%Y-%m-%d %H:%M:%S"
+        - "%Y-%m-%dT%H:%M:%S%z"
+      datetime_format: "%Y-%m-%d %H:%M:%S"
+      start_datetime:
+        type: MinMaxDatetime
+        datetime: "{{ config.start_date or day_delta(-365, format='%Y-%m-%dT%H:%M:%SZ') }}"
+        datetime_format: "%Y-%m-%dT%H:%M:%SZ"
+
+  paginator:
+    type: DefaultPaginator
+    pagination_strategy:
+      type: CustomPaginationStrategy
+      class_name: "source_mixpanel.components.EngagePaginationStrategy"
+      start_from_page: 1
+      page_size: 1000
+    page_token_option:
+      type: RequestOption
+      inject_into: request_parameter
+      field_name: page
+    page_size_option:
+      type: RequestOption
+      inject_into: request_parameter
+      field_name: page_size
+
+  # https://developer.mixpanel.com/reference/engage
+  engage_stream:
+    $ref: "#/definitions/stream_base"
+    primary_key: distinct_id
+    $parameters:
+      name: engage
+      path: 2.0/engage
+      field_path: results
+    retriever:
+      $ref: "#/definitions/retriever"
+      paginator:
+        $ref: "#/definitions/paginator"
+      record_selector:
+        $ref: "#/definitions/selector"
+        record_filter:
+          condition: "{{ record['$properties']['$last_seen'] >= stream_state.last_seen if stream_state.last_seen else true }}"
+    incremental_sync:
+      type: DatetimeBasedCursor
+      cursor_field: last_seen
+      cursor_datetime_formats:
+        - "%Y-%m-%dT%H:%M:%S"
+        - "%Y-%m-%dT%H:%M:%S%z"
+      datetime_format: "%Y-%m-%dT%H:%M:%S"
+      start_datetime:
+        type: MinMaxDatetime
+        datetime: "{{ config.start_date or day_delta(-365, format='%Y-%m-%dT%H:%M:%SZ') }}"
+        datetime_format: "%Y-%m-%dT%H:%M:%SZ"
+    transformations:
+      - class_name: "source_mixpanel.components.EngageTransformation"
+      - type: AddFields
+        fields:
+          - path:
+              - browser_version
+            value: "{{ record.browser_version | string }}"
+    schema_loader:
+      type: CustomSchemaLoader
+      class_name: "source_mixpanel.components.EngageJsonFileSchemaLoader"
+      file_path: "./source_mixpanel/schemas/{{ parameters['name'] }}.json"
+
+  cohort_members_stream:
+    $ref: "#/definitions/engage_stream"
```

Comment:
I'm sorry if I'm not allowed to share a review here but if I understand well the code, the unique key will be the distinct_id which won't work. 
A user / profile can be in multiple cohorts. I propose to create a composite key based on cohort_id + distinct_id

Reply:
Ah, thank you for posting this!

Reply:
@RobinHerzog looks like you are right. The key should be  cohort_id + distinct_id but lets handle this in separate PR

Reply:
@midavadim do you have a follow up PR? This is a pretty big problem as it can lead to missing records in the destination and I'm worried this will slip through if we don't have a quick follow up